diff --git a/CMake/install_config.cmake b/CMake/install_config.cmake
index 9f56666117..def27cb458 100644
--- a/CMake/install_config.cmake
+++ b/CMake/install_config.cmake
@@ -47,5 +47,3 @@ install(FILES "${CMAKE_CURRENT_BINARY_DIR}/realsense2ConfigVersion.cmake"
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/config/realsense2.pc"
         DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
 )
-
-install(CODE "execute_process(COMMAND ldconfig)")
