diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -143,13 +143,5 @@
-# checkout submodule if not already checked out or clone repo if no .gitmodules file
-rocprofiler_checkout_git_submodule(
-    RECURSIVE
-    RELATIVE_PATH external/cereal
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    REPO_URL https://github.com/jrmadsen/cereal.git
-    REPO_BRANCH "rocprofiler")
+find_package(cereal REQUIRED CONFIG GLOBAL)
+target_link_libraries(rocprofiler-sdk-cereal INTERFACE cereal::cereal)
 
 target_compile_definitions(rocprofiler-sdk-cereal
                            INTERFACE $<BUILD_INTERFACE:CEREAL_THREAD_SAFE=1>)
-target_include_directories(
-    rocprofiler-sdk-cereal
-    INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cereal/include>)
